sys/vm_map.c

```diff
@@ -245,9 +241,8 @@ vm_map_t *vm_map_clone(vm_map_t *map) {
 
   rw_scoped_enter(&map->rwlock, RW_READER);
 
-  /* Temporarily switch to the new map, so that we may write contents. Note that
-     it's okay if we get preempted - the working vm map will be restored on
-     context switch. */
+  /* Temporarily switch to the new map, so that we may write contents. */
+  critical_enter();
```

Comment:
Is really `critical_enter()` required here? Can we make weaker assumptions?

Perhaps `thread_self()->proc->p_nthreads == 1` assertion is just enough to perform this step correctly?